### PR TITLE
Remove filePath and etag from `GetRestoredFileContent` 

### PR DESCRIPTION
**Pull request by yufeih**
- Don't read the whole file content just to retrieve etag, because some content may be huge like xrefmap.
- Remove filePath and etag from `GetRestoredFileContent` since they are not used outside of restore
- Reworked `RestoreMap.Create` to lift `try...catch` logic outside.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4802)

### CODE REVIEW

File: src/docfx/restore/lock/DependencyLock.cs [hunk 1]
```diff
@@ -51,12 +51,7 @@ public static DependencyLockModel Load(string docset, SourceInfo<string> depende
                 }
             }
 
-            var (_, content, _) = RestoreMap.GetRestoredFileContent(docset, dependencyLockPath);
-
-            if (string.IsNullOrEmpty(content))
```
Comment (yufeih) [inline src/docfx/restore/lock/DependencyLock.cs]: this shouldn't use `string.IsNullOrEmpty`

Comment (OsmondJiang) [reply src/docfx/restore/lock/DependencyLock.cs]: so to fix this line, changed so many files..? can we split the other changes to other PRs?

Comment (yufeih) [reply src/docfx/restore/lock/DependencyLock.cs]: sure

File: src/docfx/restore/RestoreMap.cs [hunk 2]
```diff
@@ -88,28 +87,31 @@ public static (string localPath, string content, string etag) GetRestoredFileCon
...
+                        return File.ReadAllText(fullPath);
                     }
                 }
 
                 throw Errors.FileNotFound(url).ToException();
             }
 
-            var (content, etag) = TryGetRestoredFileContent(url);
-            if (string.IsNullOrEmpty(content))
+            var filePath = RestoreFile.GetRestoreContentPath(url);
```
Comment (OsmondJiang) [inline src/docfx/restore/RestoreMap.cs]: put it inside ProcessMutex?

**Pull request merged**